From b04c7d2c48f7f608db3a8e1c2a0e418ad07ab14d Mon Sep 17 00:00:00 2001
From: Ryan Tarpine <rtarpine@chromium.org>
Date: Thu, 09 Jan 2025 10:55:26 -0800
Subject: [PATCH] [DIPS] Fix possible crash.

Check whether the result of PageSpecificContentSettings::GetForPage()
is null before using it.

Bug: 388339247
Change-Id: Ic93895ce81ed15dc1de315e55928c7e7b6b34308
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6164364
Reviewed-by: Anton Maliev <amaliev@chromium.org>
Commit-Queue: Anton Maliev <amaliev@chromium.org>
Auto-Submit: Ryan Tarpine <rtarpine@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1404304}
---

--- a/chrome/browser/dips/stateful_bounce_counter.cc
+++ b/chrome/browser/dips/stateful_bounce_counter.cc
@@ -31,9 +31,10 @@
 
 void StatefulBounceCounter::OnStatefulBounce(
     content::WebContents* web_contents) {
-  auto* pscs = content_settings::PageSpecificContentSettings::GetForPage(
-      web_contents->GetPrimaryPage());
-  pscs->IncrementStatefulBounceCount();
+  if (auto* pscs = content_settings::PageSpecificContentSettings::GetForPage(
+          web_contents->GetPrimaryPage())) {
+    pscs->IncrementStatefulBounceCount();
+  }
 }
 
 const int StatefulBounceCounter::kUserDataKey;
